src/main/java/org/kohsuke/github/Requester.java

```diff
@@ -576,11 +576,20 @@ private InputStream wrapStream(InputStream in) throws IOException {
         InputStream es = wrapStream(uc.getErrorStream());
         try {
             if (es!=null) {
+                String errorString = IOUtils.toString(es, "UTF-8");
+                // Check to see whether we hit a 403, and the message indicates the
+                // abuse handler (occurs on too many concurrent requests)
+                if (responseCode == HttpURLConnection.HTTP_FORBIDDEN && 
+                    errorString.contains("You have triggered an abuse detection mechanism.")) {
```

Comment:
My 2 cents:
I'd just check for status code and retry-after, those are the ones that are documented.
The human readable hint/text isn't quite documented.